datadog/fwprovider/resource_datadog_integration_gcp.go

```diff
@@ -0,0 +1,352 @@
+package fwprovider
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	frameworkPath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+const (
+	defaultType                    = "service_account"
+	defaultAuthURI                 = "https://accounts.google.com/o/oauth2/auth"
+	defaultTokenURI                = "https://oauth2.googleapis.com/token"
+	defaultAuthProviderX509CertURL = "https://www.googleapis.com/oauth2/v1/certs"
+	defaultClientX509CertURLPrefix = "https://www.googleapis.com/robot/v1/metadata/x509/"
+)
+
+var (
+	integrationGcpMutex sync.Mutex
+	_                   resource.ResourceWithConfigure   = (*integrationGcpResource)(nil)
+	_                   resource.ResourceWithImportState = (*integrationGcpResource)(nil)
+)
+
+type integrationGcpResource struct {
+	api  *datadogV1.GCPIntegrationApi
+	auth context.Context
+}
+
+type integrationGcpModel struct {
+	ID                             types.String `tfsdk:"id"`
+	ProjectID                      types.String `tfsdk:"project_id"`
+	PrivateKeyId                   types.String `tfsdk:"private_key_id"`
+	PrivateKey                     types.String `tfsdk:"private_key"`
+	ClientEmail                    types.String `tfsdk:"client_email"`
+	ClientId                       types.String `tfsdk:"client_id"`
+	Automute                       types.Bool   `tfsdk:"automute"`
+	HostFilters                    types.String `tfsdk:"host_filters"`
+	ResourceCollectionEnabled      types.Bool   `tfsdk:"resource_collection_enabled"`
+	CspmResourceCollectionEnabled  types.Bool   `tfsdk:"cspm_resource_collection_enabled"`
+	IsSecurityCommandCenterEnabled types.Bool   `tfsdk:"is_security_command_center_enabled"`
+}
+
+func NewIntegrationGcpResource() resource.Resource {
+	return &integrationGcpResource{}
+}
+
+func (r *integrationGcpResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
```

Comment:
Nit: I believe the `, _` can be removed? It's a `bool` that makes this a safe cast, but we ignore the bool anyways. I think we'll get an NPE on the next two lines anyway if the cast fails.